static/js/click_handlers.js

```diff
@@ -626,6 +626,11 @@ exports.initialize = function () {
                 edit_area.attr("data-prev-text", edit_area.text().trim())
                     .attr("contenteditable", true);
 
+                if (selector === ".stream-description-editable") {
+                    var sub = stream_edit.get_sub_for_target(this);
+                    edit_area.text(sub.description);
+                }
+
```

Comment:
Hmm, ideally we'd avoid adding feature-specific code in this generic library.  How about we make this a hook function and extend the `map` thing at the top to have a dict rather than just a function inside?

E.g.
```
        var map = {
            ".stream-description-editable": {
                 on_save: stream_edit.change_stream_description,
                 on_start: stream_edit.get_raw_description,
            }
            ...
        };

```

As a sidenote, a larger refactor for this component might want to include extracting this little sub-mobile to e.g. `static/js/editable_section.js` -- there's no need for it to be in `click_handlers.js`

Reply:
@timabbott, I'm not too sure of what you're suggesting I do here. For the time being, I've added an extra (*temporary*) commit to this PR, based on what I think you've asked for me to do. If this isn't what you suggested (which I strongly suggest is the case), could you please clarify what exactly you're suggesting (regarding the hook function and extension of the map object)?

Reply:
Yes, this is what I'd had in mind!  The benefit of this `map` model in general is that our generate content-editable component doesn't end up cluttered with individual bits of feature-specific conditionals, which makes it easier to reuse it for future features (or replace it with some other library, if we ever want to go that route).